lib/msf/ui/console/command_dispatcher/auxiliary.rb

```diff
@@ -98,7 +98,7 @@ def cmd_run(*args)
     # Always run passive modules in the background
     if (mod.passive || mod.passive_action?(action || mod.default_action))
       jobify = true
-    end
+    end if mod.is_a?(Msf::Module::HasActions)
```

Comment:
would you object to writing it like this instead? When a block has an if statement at the beginning _and_ the end, it is confusing what the precedence rules are.

```
if mod.is_a?(Msf::Module::HasActions) &&
    (mod.passive || mod.passive_action?(action || mod.default_action))
       jobify = true
end
```